typing/typecore.ml

```diff
@@ -4233,10 +4228,11 @@ and type_expect_
         end;
       let mode = mode_cross env ty_arg mode in
       ruem ~mode ~expected_mode {
-        exp_desc = Texp_field(record, lid, label);
+        exp_desc = Texp_field(record, lid, label,
+            Value_mode.regional_to_global_alloc expected_mode.mode
```

Comment:
It is probably worth checking the representation of the field and registering the allocation if it is a float field here. Currently we are not always locally allocating floats when we could. It might also be worth making this field optional -- although unlike the `Texp_record` case the value of `alloc_mode` should never be misleading for a `Texp_field` even when it isn't needed.